### PR TITLE
Forms for refrigerators with direct sales function

### PR DESCRIPTION
**Pull request by matteason**

### CODE REVIEW

File: src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/VendingMachinesForm.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  @FieldPrompt("The annual electricity consumption in kWh in terms of final energy consumption per year")
+  @Digits(integer = 4, fraction = 0, message = "Enter the annual energy consumption, up to 4 digits long")
+  private String annualEnergyConsumption;
+
+  @FieldPrompt("The sum of the net volumes, expressed in litres, of all compartments functioning at chilled operating temperature")
+  @Digits(integer = 4, fraction = 0, message = "Enter the total volume of chilled compartments, up to 4 digits long")
+  private String fridgeCapacity;
+
+  @FieldPrompt("The maximum measured product temperature, expressed in degrees Celcius, of the compartment(s) with chilled operating temperatures")
```
Comment (jamesbarnett91) [inline src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/VendingMachinesForm.java:29]: Is `Celcius` a british-english thing? I've not seen that before, and it's listed as `Celsius` in the Oxford dictionary, but my spelling is atrocious and I never know the different variants.

File: src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/controller/RefrigeratorsDirectSalesController.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      return getVendingMachines(bindingResult.getFieldErrors());
+    }
+    else {
+      return documentRendererService.processPdfResponse(refrigeratorsDirectSales.generateHtml(form, RefrigeratorsDirectSalesService.LEGISLATION_CATEGORY_CURRENT));
+    }
+  }
+
+  @PostMapping(value = "/vending machines", params = "mode=INTERNET")
+  @ResponseBody
+  public Object handleInternetLabelVendingMachinesSubmit(@Validated(InternetLabellingGroup.class) @ModelAttribute("form") BeverageCoolersForm form, BindingResult bindingResult) {
```
Comment (jamesbarnett91) [inline src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/controller/RefrigeratorsDirectSalesController.java:128]: Should take in a `VendingMachinesForm`

File: src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/VendingMachinesForm.java (hunk 1 above)
Comment (matteason) [reply src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/VendingMachinesForm.java:29]: No my bad, it's a typo. Anders Celsius would be ashamed of me

File: src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/DisplayCabinetsForm.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  @FieldPrompt("The annual electricity consumption in kWh in terms of final energy consumption per year")
+  @Digits(integer = 4, fraction = 0, message = "Enter the annual energy consumption, up to 4 digits long")
+  private String annualEnergyConsumption;
+
+  @FieldPrompt("Does this model have any display areas functioning at chilled operating temperature?")
+  @NotNull(message = "Select whether this model has chilled display areas")
+  private Boolean chilledCompartment;
+
+  @FieldPrompt("The sum of the display areas, expressed in square metres, of all compartments functioning at chilled operating temperature")
+  @Digits(groups = FridgeGroup.class, integer = 4, fraction = 2, message = "Enter the total volume of chilled compartments, up to 4 digits long with up to 2 decimal places")
```
Comment (jamesbarnett91) [inline src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/DisplayCabinetsForm.java:30]: Validation message talks about volume but the question text is display area

File: src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/DisplayCabinetsForm.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  @FieldPrompt("The lowest temperature of the coldest M-package, or the highest minimum temperature of all M-packages, expressed in degrees Celcius of the compartment(s) with chilled operating temperatures")
+  @Digits(groups = FridgeGroup.class, integer = 2, fraction = 0, message = "Enter lowest temperature of the coldest M-package, or the highest minimum temperature of all M-packages in the chilled compartments, up to 2 digits long")
+  private String fridgeMinTemp;
+
+  @FieldPrompt("Does this model have any display areas functioning at frozen operating temperature?")
+  @NotNull(message = "Select whether this model has frozen display areas")
+  private Boolean frozenCompartment;
+
+  @FieldPrompt("The sum of the display areas, expressed in square metres, of all display areas functioning at frozen operating temperature")
+  @Digits(groups = FreezerGroup.class, integer = 4, fraction = 2, message = "Enter the total volume of chilled compartments, up to 4 digits long with up to 2 decimal places")
```
Comment (jamesbarnett91) [inline src/main/java/uk/gov/beis/els/categories/refrigeratorsdirectsales/model/DisplayCabinetsForm.java:46]: Same as above, also should be the frozen compartment

**Review by jamesbarnett91 — CHANGES_REQUESTED** [review]

**Review by jamesbarnett91 — APPROVED** [review]

**Pull request merged**